### PR TITLE
docs: Improve plugin template docs. Fixes #7683

### LINKED ISSUE #7683: Improve plugin templates docs
* Do plugin templates only support Python? It's not clear in the docs:
  * https://github.com/argoproj/argo-workflows/blob/master/docs/executor_plugins.md
  * https://github.com/argoproj/argo-workflows/blob/master/docs/plugins.md
* Does a plugin failure cause a workflow failure? If so does it support the [continueOn](https://github.com/argoproj/argo-workflows/blob/4e450e250168e6b4d51a126b784e90b11a0162bc/examples/continue-on-fail.yaml) feature?
* How do users securely manage secrets needed for a plugin? Such as a Slack token in your example.
* Should we have an example in https://github.com/argoproj/argo-workflows/tree/master/examples for this feature? 
* Is the community going to maintain any common plugins such as a Slack plugin? 
* How do plugins interact with features such as resource requests/limits , pod disruption budgets, retry on the workflow side and activeDeadlineSeconds? It's a template so all template/step features should work right?
* What are the use cases for the aysnc / requeue feature? Is the 2 minutes configurable?

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>
Fixes #7683

Don't bother creating a PR until you've done this:

* [ ] Run `make pre-commit -B` to fix codegen, lint, and commit message problems.

Create your PR as a draft.

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it
  does not need to pass.
* Once required tests have passed, you can make it "Ready for review".
* Say how you tested your changes. If you changed the UI, attach screenshots.

Tips:

* If changes were requested, and you've made them, then dismiss the review to get it looked at again.
* Add you organization to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md) if you like.
* You can ask for help!

### CODE REVIEW

**Comment by alexec** [comment]
@blkperl

File: docs/executor_plugins.md [hunk 1]
```diff
@@ -1,21 +1,74 @@
...
-3. Configuration so the controller can discover the plugin.
+1. Plugins enabled (see above).
+2. A HTTP server that will be run as a sidecar to the main container and will respond to RPC HTTP requests from the
+   executor with [this API contract](executor_swagger.md).
+3. A `plugin.yaml` configuration file, that is turned into a config map so the controller can discover the plugin.
+
+A template executor plugin services HTTP POST requests on `/api/v1/template.execute`:
+
+```shell
+curl http://localhost:4355//api/v1/template.execute -d \
```
Comment (blkperl) [inline docs/executor_plugins.md:43]: Is the extra slash a typo?

File: docs/executor_plugins.md [hunk 2]
```diff
@@ -119,35 +178,106 @@ spec:
...
+  sidecar:
+    container:
+      env:
+        - name: URL
+          valueFrom:
+            secretKeyRef:
+              name: slack-executor-plugin
+              key: URL
+```
+
```
Comment (blkperl) [inline docs/executor_plugins.md:206]:
```suggestion
Refer to the [Kubernetes Secret documentation] for secret best practices and security considerations

[Kubernetes Secret documentation]: https://kubernetes.io/docs/concepts/configuration/secret
```

**Review by blkperl — APPROVED** [review]
Looks good, I will try to reserve some time next week to experiment with the feature.

File: docs/executor_plugins.md (hunk 1 above)
Comment (alexec) [reply docs/executor_plugins.md:43]: no, this is so next line is escaped

File: docs/executor_plugins.md (hunk 2 above)
Comment (alexec) [reply docs/executor_plugins.md:206]: I'll add this

**Review by sarabala1979 — APPROVED** [review]

**Pull request merged**